aiokafka/fetcher.py

```diff
@@ -0,0 +1,579 @@
+            fetch_min_bytes (int): Minimum amount of data the server should
+                return for a fetch request, otherwise wait up to
+                fetch_max_wait_ms for more data to accumulate. Default: 1.
+            fetch_max_wait_ms (int): The maximum amount of time in milliseconds
+                the server will block before answering the fetch request if
+                there isn't sufficient data to immediately satisfy the
+                requirement given by fetch_min_bytes. Default: 500.
+            max_partition_fetch_bytes (int): The maximum amount of data
+                per-partition the server will return. The maximum total memory
+                used for a request = #partitions * max_partition_fetch_bytes.
+                This size must be at least as large as the maximum message size
+                the server allows or else it is possible for the producer to
+                send messages larger than the consumer can fetch. If that
+                happens, the consumer can get stuck trying to fetch a large
+                message on a certain partition. Default: 1048576.
+            check_crcs (bool): Automatically check the CRC32 of the records
+                consumed. This ensures no on-the-wire or on-disk corruption to
+                the messages occurred. This check adds some overhead, so it may
+                be disabled in cases seeking extreme performance. Default: True
+            fetcher_timeout (float): number of second to poll necessity to send
+                next fetch request. Default: 0.1
+        """
+        self._client = client
+        self._loop = loop
+        self._key_deserializer = key_deserializer
+        self._value_deserializer = value_deserializer
+        self._fetch_min_bytes = fetch_min_bytes
+        self._fetch_max_wait_ms = fetch_max_wait_ms
+        self._max_partition_fetch_bytes = max_partition_fetch_bytes
+        self._check_crcs = check_crcs
+        self._fetcher_timeout = fetcher_timeout
+        self._subscriptions = subscriptions
+        self._records = collections.defaultdict(list)
+        self._unauthorized_topics = set()
+        self._offset_out_of_range_partitions = dict()  # {partition: offset}
+        self._record_too_large_partitions = dict()  # {partition: offset}
+        self._fetch_task = None
+        self._iterator = None
+
+    @asyncio.coroutine
+    def close(self):
+        if self._fetch_task:
+            self._fetch_task.cancel()
+            try:
+                yield from self._fetch_task
+            except asyncio.CancelledError:
+                pass
+            self._fetch_task = None
+
+    def _init_fetches(self):
+        """Send FetchRequests asynchronously for all assigned partitions.
+        """
+        if self._fetch_task is None:
+            self._fetch_task = ensure_future(
+                self._fetch_requests_routine(), loop=self._loop)
+
+    @asyncio.coroutine
+    def _fetch_requests_routine(self):
```

Comment:
We can use subscription listener for this. Lets remove it from client's API. If needed we can add a more flexible event scheme.